AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginRESTIAMTests/RESTWithIAMIntegrationTests.swift

```diff
@@ -38,193 +38,122 @@ class RESTWithIAMIntegrationTests: XCTestCase {
+    func testGetAPIWithQueryParamsSuccess() async throws {
         let request = RESTRequest(path: "/items",
                                   queryParameters: [
                                     "user": "hello@email.com",
                                     "created": "2021-06-18T09:00:00Z"
                                   ])
-        _ = Amplify.API.get(request: request) { event in
-            switch event {
-            case .success(let data):
-                let result = String(decoding: data, as: UTF8.self)
-                print(result)
-                completeInvoked.fulfill()
-            case .failure(let error):
-                XCTFail("Unexpected .failed event: \(error)")
-            }
-        }
-
-        await waitForExpectations(timeout: TestCommonConstants.networkTimeout)
+        let data = try await Amplify.API.get(request: request)
+        let result = String(decoding: data, as: UTF8.self)
+        print(result)
     }
 
-    func testGetAPIWithEncodedQueryParamsSuccess() async {
-        let completeInvoked = expectation(description: "request completed")
+    // TODO: Should not be HTTPStatusError
+    func testGetAPIWithEncodedQueryParamsSuccess() async throws {
         let request = RESTRequest(path: "/items",
                                   queryParameters: [
                                     "user": "hello%40email.com",
                                     "created": "2021-06-18T09%3A00%3A00Z"
                                   ])
-        _ = Amplify.API.get(request: request) { event in
-            switch event {
-            case .success(let data):
-                let result = String(decoding: data, as: UTF8.self)
-                print(result)
-                completeInvoked.fulfill()
-            case .failure(let error):
-                XCTFail("Unexpected .failed event: \(error)")
-            }
-        }
-
-        await waitForExpectations(timeout: TestCommonConstants.networkTimeout)
+        let data = try await Amplify.API.get(request: request)
+        let result = String(decoding: data, as: UTF8.self)
+        print(result)
     }
 
-    func testPutAPISuccess() async {
-        let completeInvoked = expectation(description: "request completed")
+    func testPutAPISuccess() async throws {
         let request = RESTRequest(path: "/items")
-        _ = Amplify.API.put(request: request) { event in
-            switch event {
-            case .success(let data):
-                let result = String(decoding: data, as: UTF8.self)
-                print(result)
-                completeInvoked.fulfill()
-            case .failure(let error):
-                XCTFail("Unexpected .failed event: \(error)")
-            }
-        }
-
-    await waitForExpectations(timeout: TestCommonConstants.networkTimeout)
+        let data = try await Amplify.API.put(request: request)
+        let result = String(decoding: data, as: UTF8.self)
+        print(result)
```

Comment:
Can we add meaningful assertions here? We can add these in a follow up if the changes in this PR are to get the tests compiling. Just want to make sure we're not saying this is an acceptable end state of these tests.